include/oneapi/dpl/pstl/utils.h

```diff
@@ -627,9 +655,20 @@ __dpl_signbit(const _T& __x)
     return (__x & __mask) != 0;
 }
 
-template <typename _Acc, typename _Size1, typename _Value, typename _Compare>
+// Adapts __pstl_lower_bound and other related functions to use with host backends
+struct _SubscriptAdapter
+{
+    template <typename _Iterator>
+    decltype(auto)
+    operator[](_Iterator __it) const
```

Comment:
Why do we need a weird operator that pretends to do indexing but takes an iterator as the argument? To me, it screams as an improper workaround for an architecture problem.

Reply:
The PR created: https://github.com/uxlfoundation/oneDPL/pull/2409